src/morale.cpp

```diff
@@ -637,6 +660,30 @@ void player_morale::update_stylish_bonus()
     set_permanent( MORALE_PERM_FANCY, bonus );
 }
 
+void player_morale::update_squeamish_penalty()
+{
+    int penalty = 0;
+
+    if( squeamish ) {
+        const auto bp_pen = [ this ]( body_part bp, int penalty ) -> int {
+            return (
+                body_parts[bp].filthy > 0 ||
+                body_parts[opposite_body_part( bp )].filthy > 0 ) ? penalty : 0;
+        };
+        penalty = std::min (bp_pen( bp_torso,  10 ) +
+                            bp_pen( bp_head,   7 ) +
+                            bp_pen( bp_eyes,   5 ) +
+                            bp_pen( bp_mouth,  5 ) +
+                            bp_pen( bp_leg_l,  5 ) +
+                            bp_pen( bp_leg_r,  5 ) +
+                            bp_pen( bp_foot_l, 3 ) +
+                            bp_pen( bp_foot_r, 3 ) +
+                            bp_pen( bp_hand_l, 3 ) +
+                            bp_pen( bp_hand_r, 3 ), 50 );
```

Comment:
You probably don't need `min()` here. The penalties in total are less than the limit. Also, penalties for mouth, eyes and head IMO should be the highest ones. I'd rearrange them.